packages/gatsby-graphiql-explorer/src/app/snippets.js

```diff
@@ -69,4 +69,34 @@ export default ComponentName
 `,
 }
 
-export default [pageQuery, staticHook, staticQuery]
+const createPages = {
+  name: `createPages`,
+  language: `JavaScript`,
+  codeMirrorMode: `javascript`,
+  options: [],
+  generate: arg => `const path = require(\`path\`)
+
+exports.createPages = async ({ graphql, actions }) => {
+  const { createPage } = actions
+
+  const result = await graphql(\`
+${getQuery(arg, 4)}
+  \`)
+
+  result.data.${
+    arg.operationDataList[0].operationDefinition.selectionSet.selections[0].name
+      .value
+  }.edges.forEach(({ node }) => {
+    createPage({
+      path: NODE_SLUG,
+      component: path.resolve(\`PATH/TO/TEMPLATE.js\`),
```

Comment:
It's a little thing but sometimes matters for perf but if you could move the path.resolve out of the forEach so it's only called once.

Reply:
Updated